nix/commands/test.sh

```diff
@@ -3,9 +3,30 @@
 set -e
 
 basedir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+source "$basedir/__load-config.sh"
 
-source $basedir/__load-config.sh
+# to run the tests, you'll first need to build the test modules
+# against the same version of Node.js that was built as part of
+# the build process.
+declare -r build_dir="${ELECTRON_GN_ROOT}/src/out/${ELECTRON_OUT_DIR}"
+declare -r node_headers_dir="${build_dir}/gen/node_headers"
+declare -r electron_spec_dir="${ELECTRON_GN_ROOT}/src/electron/spec"
 
-cd $ELECTRON_GN_ROOT/src/electron
+# does it need to be rebuilt?
+if [ ! -d "${node_headers_dir}" ]; then
+  declare -r node_headers_need_rebuild='yes'
+elif [ "${electron_spec_dir}/package.json" -nt "${node_headers_dir}" ]; then
```

Comment:
Neat trick, I can't think of a case where this will false-negative but hey, it's better than what we had before (nothing) 😄